src/plugins/replyPingControl/index.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2023 Vendicated, Mr Diamond, ant0n and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { definePluginSettings } from "@api/Settings";
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+import { MessageStore, UserStore } from "@webpack/common";
+import { MessageJSON, UserJSON } from "discord-types/general";
+
+export const settings = definePluginSettings({
+    alwaysPingOnReply: {
+        type: OptionType.BOOLEAN,
+        description: "Always get pinged when someone replies to your messages",
+        default: false,
+    }
+});
+
+export default definePlugin({
+    name: "ReplyPingControl",
+    description: "Control whether to always or never get pinged on message replies",
+    authors: [Devs.ant0n, Devs.MrDiamond],
+    settings,
+
+    patches: [
+        {
+            find: "_channelMessages",
+            replacement: {
+                match: /receiveMessage\((\i)\)\{/,
+                replace: "$&$self.modifyMentions($1);"
+            }
+        }
+    ],
+
+    modifyMentions(message: MessageJSON) {
+
+        if (!this.isReplyToCurrentUser(message)) return;
+
+        if (settings.store.alwaysPingOnReply) {
+            if (!message.mentions.some(mention => mention.id === UserStore.getCurrentUser().id)) {
+                message.mentions.push(this.getCurrentUserMention());
+            }
+        } else {
+
+            message.mentions = message.mentions.filter(mention => mention.id !== UserStore.getCurrentUser().id);
+        }
+    },
+
+    isReplyToCurrentUser(message: MessageJSON) {
+        if (!message.message_reference) return false;
+        const repliedMessage = MessageStore.getMessage(message.channel_id, message.message_reference.message_id);
+        return repliedMessage && repliedMessage.author.id === UserStore.getCurrentUser().id;
+    },
+
+    getCurrentUserMention() {
+        return UserStore.getCurrentUser() as unknown as UserJSON;
+    }
```

Comment:
this formatting is kinda weird, I'd also simplify it like:

```suggestion
    patches: [{
        find: "_channelMessages",
        replacement: {
            match: /receiveMessage\((\i)\)\{/,
            replace: "$&$self.modifyMentions($1);"
        }
    }],

    modifyMentions(message: MessageJSON) {
        const user = UserStore.getCurrentUser();

        if (this.getRepliedMessage(message)?.author.id !== user.id)
            return;

        if (!settings.store.alwaysPingOnReply)
            message.mentions = message.mentions.filter(mention => mention.id !== user.id);
        else if (!message.mentions.some(mention => mention.id === user.id))
            message.mentions.push(user as any);
    },

    getRepliedMessage(message: MessageJSON) {
        const ref = message.message_reference;
        return ref && MessageStore.getMessage(ref.channel_id, ref.message_id);
    },
```

Reply:
The logic worked fine before, I don't think it should be changed. This makes it a little harder to read imo

Reply:
This was done more for the sake of verbosity really, works fine. I'll test both your suggestions in a sec.

Reply:
I'll merge it for fewer calls and no need to import UserJSON